emukit/quadrature/kernels/quadrature_matern32.py

```diff
@@ -0,0 +1,170 @@
+
+        :param x1: remaining argument of the once integrated kernel, shape (n_point N, input_dim)
+        :returns: kernel mean at location x1, shape (N, 1)
+        """
+        return self.qK(x1).T
+
+    def qKq(self) -> float:
+        """ProductMatern32 kernel integrated over both arguments x1 and x2.
+
+        :returns: double integrated kernel
+        """
+        raise NotImplementedError
+
+    def dqK_dx(self, x2: np.ndarray) -> np.ndarray:
+        """
+        gradient of the kernel mean (integrated in first argument) evaluated at x2
+        :param x2: points at which to evaluate, shape (n_point N, input_dim)
+        :return: the gradient with shape (input_dim, N)
+        """
+        raise NotImplementedError
+
+    def dKq_dx(self, x1: np.ndarray) -> np.ndarray:
+        """
+        gradient of the kernel mean (integrated in second argument) evaluated at x1
+        :param x1: points at which to evaluate, shape (n_point N, input_dim)
+        :return: the gradient with shape (N, input_dim)
+        """
+        return self.dqK_dx(x1).T
+
+
+class QuadratureProductMatern32LebesgueMeasure(QuadratureMatern32):
+    """A Matern32 product kernel with integrability over the standard Lebesgue measure.
+
+    Can only be used with finite integral bounds.
+    """
+
+    def __init__(
+        self, matern_kernel: IProductMatern32, integral_bounds: List[Tuple[float, float]], variable_names: str = ""
+    ) -> None:
+        """
+        :param matern_kernel: Standard emukit matern32-kernel.
+        :param integral_bounds: defines the domain of the integral. List of D tuples, where D is the dimensionality
+        of the integral and the tuples contain the lower and upper bounds of the integral
+        i.e., [(lb_1, ub_1), (lb_2, ub_2), ..., (lb_D, ub_D)].
+        :param variable_names: The (variable) name(s) of the integral.
+        """
+        super().__init__(
+            matern_kernel=matern_kernel, integral_bounds=integral_bounds, measure=None, variable_names=variable_names
+        )
+
+    def qK(self, x2: np.ndarray, skip: List[int] = None) -> np.ndarray:
+        """Matern32 prodct kernel with the first component integrated out aka kernel mean.
+
+        :param x2: remaining argument of the once integrated kernel, shape (n_point N, input_dim)
+        :param skip: Skip those dimensions from product.
+        :returns: kernel mean at location x2, shape (1, N)
+        """
+        if skip is None:
+            skip = []
+
+        qK = np.ones(x2.shape[0])
+        for dim in range(x2.shape[1]):
+            if dim in skip:
+                continue
+            qK *= self._qK_1d(x=x2[:, dim], domain=self.integral_bounds.bounds[dim], ell=self.lengthscales[dim])
+        return qK[None, :] * self.variance
+
+    def qKq(self) -> float:
+        """Matern32 kernel integrated over both arguments x1 and x2
+
+        :returns: double integrated kernel
```

Comment:
Is that just a final result of the integration? The wording feels a bit unclear

Reply:
Not a final result. Simply kernel integrated in both arguments. It can be used in various ways.